examples/src/chains/map_reduce_lcel.ts

```diff
@@ -0,0 +1,163 @@
+import { BaseCallbackConfig } from "langchain/callbacks";
+import {
+  _collapseDocs,
+  _splitListOfDocs,
+} from "langchain/chains/combine_documents/reduce";
+import { ChatOpenAI } from "langchain/chat_models/openai";
+import { Document } from "langchain/document";
+import { PromptTemplate } from "langchain/prompts";
+import { StringOutputParser } from "langchain/schema/output_parser";
+import { formatDocument } from "langchain/schema/prompt_template";
+import {
+  RunnableLambda,
+  RunnableMap,
+  RunnablePassthrough,
+  RunnableSequence,
+} from "langchain/schema/runnable";
+
+// Initialize the OpenAI model
+const model = new ChatOpenAI({});
+
+// Define prompt templates for document formatting, summarizing, collapsing, and combining
+const documentPrompt = PromptTemplate.fromTemplate("{pageContent}");
+const summarizePrompt = PromptTemplate.fromTemplate(
+  "Summarize this content:\n\n{context}"
+);
+const collapsePrompt = PromptTemplate.fromTemplate(
+  "Collapse this content:\n\n{context}"
+);
+const combinePrompt = PromptTemplate.fromTemplate(
+  "Combine these summaries:\n\n{context}"
+);
+
+// Define a util function to allow function invocation with preset arguments
+function partial<F extends (...args: any[]) => any>(
+  fn: F,
+  ...partialArgs: any[]
+): (...args: any[]) => ReturnType<F> {
+  return (...args: any[]): ReturnType<F> => fn(...partialArgs, ...args);
```

Comment:
Can't we just use `.bind`?

Reply:
Was able to refactor and scrap this util